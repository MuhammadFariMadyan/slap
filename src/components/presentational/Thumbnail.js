import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import "typeface-roboto";

const Container = styled.svg`
  background: white;
  border: 1px solid #ddd;
  cursor: pointer;
`;

const Text = styled.text`
  font-family: Roboto;
  color: black;
`;

const Thumbnail = ({ width, contents, onClick }) => {
  const baseWidth = 640;
  const baseHeight = 480;
  const baseFontSize = 32;
  const height = (baseHeight / baseWidth) * width;
  const scale = width / baseWidth;
  const fontSize = baseFontSize * scale;

  return (
    <Container width={width} height={height} onClick={onClick}>
      {contents.map(content => {
        const { position } = content;
        const x = position.x * scale;
        const y = position.y * scale;

        return (
          <Text key={content.id} x={x} y={y} fontSize={fontSize}>
            {content.content}
          </Text>
        );
      })}
    </Container>
  );
};

Thumbnail.propTypes = {
  width: PropTypes.number,
  contents: PropTypes.array,
  onClick: PropTypes.func
};

Thumbnail.defaultProps = {
  width: 240,
  contents: []
};

export default Thumbnail;
