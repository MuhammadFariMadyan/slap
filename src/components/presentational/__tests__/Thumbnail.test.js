import React from "react";
import { shallow } from "enzyme";

import Thumbnail from "../Thumbnail";

describe("Thumbnail", () => {
  const setup = propOverrides => {
    const props = {
      width: 240,
      contents: [
        {
          id: "content-1",
          slideID: "slide-1",
          projectID: "project-1",
          type: "text",
          content: "Hello World",
          position: {
            x: 32,
            y: 32
          }
        },
        {
          id: "content-2",
          slideID: "slide-1",
          projectID: "project-1",
          type: "text",
          content: "Stay hungry",
          position: {
            x: 128,
            y: 96
          }
        }
      ],
      onClick: jest.fn(),
      ...propOverrides
    };

    const wrapper = shallow(<Thumbnail {...props} />);

    return {
      wrapper,
      props
    };
  };

  it("should renders default", () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it("should renders without contents", () => {
    const { wrapper } = setup({ contents: [] });

    expect(wrapper).toMatchSnapshot();
  });

  it("should calls 'onClick' callback while clicked", () => {
    const { wrapper, props } = setup();

    wrapper.simulate("click");

    expect(props.onClick).toBeCalled();
  });
});
