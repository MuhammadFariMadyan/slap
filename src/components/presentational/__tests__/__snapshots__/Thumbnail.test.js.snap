// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thumbnail should renders default 1`] = `
<styled.svg
  height={180}
  onClick={[Function]}
  width={240}
>
  <styled.text
    fontSize={12}
    key="content-1"
    x={12}
    y={12}
  >
    Hello World
  </styled.text>
  <styled.text
    fontSize={12}
    key="content-2"
    x={48}
    y={36}
  >
    Stay hungry
  </styled.text>
</styled.svg>
`;

exports[`Thumbnail should renders without contents 1`] = `
<styled.svg
  height={180}
  onClick={[Function]}
  width={240}
/>
`;
